lib/compat/wordpress-6.7/class-gutenberg-rest-posts-controller-6-7.php

```diff
@@ -0,0 +1,1139 @@
+								'description' => __( 'Excerpt for the post, as it exists in the database.' ),
+								'type'        => 'string',
+								'context'     => array( 'edit' ),
+							),
+							'rendered'  => array(
+								'description' => __( 'HTML excerpt for the post, transformed for display.' ),
+								'type'        => 'string',
+								'context'     => array( 'view', 'edit', 'embed' ),
+								'readonly'    => true,
+							),
+							'protected' => array(
+								'description' => __( 'Whether the excerpt is protected with a password.' ),
+								'type'        => 'boolean',
+								'context'     => array( 'view', 'edit', 'embed' ),
+								'readonly'    => true,
+							),
+						),
+					);
+					break;
+
+				case 'thumbnail':
+					$schema['properties']['featured_media'] = array(
+						'description' => __( 'The ID of the featured media for the post.' ),
+						'type'        => 'integer',
+						'context'     => array( 'view', 'edit', 'embed' ),
+					);
+					break;
+
+				case 'comments':
+					$schema['properties']['comment_status'] = array(
+						'description' => __( 'Whether or not comments are open on the post.' ),
+						'type'        => 'string',
+						'enum'        => array( 'open', 'closed' ),
+						'context'     => array( 'view', 'edit' ),
+					);
+					$schema['properties']['ping_status']    = array(
+						'description' => __( 'Whether or not the post can be pinged.' ),
+						'type'        => 'string',
+						'enum'        => array( 'open', 'closed' ),
+						'context'     => array( 'view', 'edit' ),
+					);
+					break;
+
+				case 'page-attributes':
+					$schema['properties']['menu_order'] = array(
+						'description' => __( 'The order of the post in relation to other posts.' ),
+						'type'        => 'integer',
+						'context'     => array( 'view', 'edit' ),
+					);
+					break;
+
+				case 'post-formats':
+					// Get the native post formats and remove the array keys.
+					$formats = array_values( get_post_format_slugs() );
+
+					$schema['properties']['format'] = array(
+						'description' => __( 'The format for the post.' ),
+						'type'        => 'string',
+						'enum'        => $formats,
+						'context'     => array( 'view', 'edit' ),
+					);
+					break;
+
+				case 'custom-fields':
+					$schema['properties']['meta'] = $this->meta->get_field_schema();
+					break;
+
+			}
+		}
+
+		if ( 'post' === $this->post_type ) {
+			$schema['properties']['sticky'] = array(
+				'description' => __( 'Whether or not the post should be treated as sticky.' ),
+				'type'        => 'boolean',
+				'context'     => array( 'view', 'edit' ),
+			);
+		}
+
+		$schema['properties']['template'] = array(
+			'description' => __( 'The theme file to use to display the post.' ),
+			'type'        => 'string',
+			'context'     => array( 'view', 'edit' ),
+			'arg_options' => array(
+				'validate_callback' => array( $this, 'check_template' ),
+			),
+		);
+
+		$schema['properties']['format'] = array(
```

Comment:
The change starts here.
I am trying to allow both a string or an array, but only the valid post format slugs.
It might be fine to just allow the string?

Reply:
Update: There is definitely something wrong, because if I don't comment this out, I am not able to assign formats to posts in the block editor.